### ISSUE TITLE
can add a mind map feature?

### ISSUE THREAD

**Issue by jan-bar** [zadam/trilium#3103]
### Describe feature

Since I'm not familiar with js, I found a [mind map project](https://github.com/hizzgdev/jsmind) and used trilium to configure a read-only mind map note. 

This library supports editing nodes, but I feel that I need to use the front-end and back-end api functions of trilium, which I am not familiar with, so I hope the official add the function of mind map notes.

![image](https://user-images.githubusercontent.com/14202095/187618661-034e1d52-ecc6-44d9-b39d-dd05b318b629.png)

Here are my exported notes：[mind.zip](https://github.com/zadam/trilium/files/9460016/mind.zip)

The official css needs to annotate this configuration before it can be displayed to trilium normally

![image](https://user-images.githubusercontent.com/14202095/187619565-386ea5d5-7ca7-4143-b70f-61aae42ada08.png)

By reading the wiki need to add `#appCss` tag for css

![image](https://user-images.githubusercontent.com/14202095/187619750-39783ce7-b00f-44ca-b0c2-6d2be471536c.png)

The reason why it is read-only is because I have written data here.

![image](https://user-images.githubusercontent.com/14202095/187636397-9933f79f-07af-42a2-91f6-b980fbb6eb37.png)

![image](https://user-images.githubusercontent.com/14202095/187620116-d0350f00-4060-4562-a345-0870f322937b.png)

![image](https://user-images.githubusercontent.com/14202095/187620176-14619799-1c5d-4baa-9b07-1c2ac1177179.png)

Now I convert xmind to the required json format data in the following way, and get a read-only mind map note in the above way.  `go run main.go -s my.xmind -d my.json`

```go
package main

import (
	"flag"
	"os"

	"github.com/jan-bar/xmind"
)

func main() {
	xm := flag.String("s", "", "src xmind file")
	js := flag.String("d", "", "dst json file")
	flag.Parse()

	st, err := xmind.LoadFile(*xm)
	if err != nil {
		panic(err)
	}

	var data []byte
	err = xmind.SaveCustom(st.Topics[0], "id", "topic", "parentid", "isroot,1", &data, xmind.CustomIncrId())
	if err != nil {
		panic(err)
	}

	err = os.WriteFile(*js, data, os.ModePerm)
	if err != nil {
		panic(err)
	}
}
```

### Additional Information

I switched from other note-taking software to trilium, because the previous note-taking software has a mind map function, so I am not used to it, but the trilium function is very useful, especially script notes, but my JavaScript learning is not very deep, I hope the author can refer to , add mind map notes, thank you very much.

**Comment by sigaloid** [comment]
Something like this is already similar with Mermaid graphs: https://github.com/zadam/trilium/pull/2187 

Have you tried implementing a mind map within a Mermaid graph?

**Comment by jan-bar** [comment]
@sigaloid  This is indeed a solution, but it cannot be edited in a faster way on the node. I feel like I can write a way to convert from Xmind to Mermaid graphs.

**Comment by banjuer** [comment]
@jan-bar hi bro, are there trilium plugins available

**Comment by jan-bar** [comment]
> @jan-bar hi bro, are there trilium plugins available

No, you can refer to the project above and the plan I gave. I don’t know much about JavaScript.

**Comment by banjuer** [comment]
@jan-bar hi bro, i found one, try this: https://github.com/waterovo/trilium-jsmind
awesome

**Comment by eliandoran** [comment]
Mind map functionality was introduced into [TriliumNext](https://github.com/TriliumNext/Notes) using [MindElixir](https://mind-elixir.com/). Feel free to check it out on the latest stable or beta release (where we have a configuration pane as well).
